### ISSUE TITLE
broken .md links

### ISSUE THREAD

**Issue by tabcat** [ipld/ipld#176]
There are many links inside markdown files linking to other markdown files that are broken on github.

They are all only broken by a little bit; the links are just missing the `.md` extension. Here is an example of a [broken link inside the repo](https://github.com/ipld/ipld/blob/master/specs/selectors/index.md#fixtures). Many of these exist.

Happy to make a pr and comment of a script that handled repairing the links.

**Comment by tabcat** [comment]
Was wondering if there was a reason for this. The links work on the ipld site. Adding `.md` to the paths result with an ipfs error since there is no link named FAQ.md under [bafybeidpga2xwiiseib3tcsp33wodu3itunwj2su624yygdx6av3s4f5f4](https://explore.ipld.io/#/explore/bafybeidpga2xwiiseib3tcsp33wodu3itunwj2su624yygdx6av3s4f5f4). For them to work on github and the website both would need to be changed to `.md`, but there may have been a reason for this in the first place.

**Comment by rvagg** [comment]
Yeah, this annoys me a lot, I _think_ it's because eleventy, the static site generator, needs them in this broken state, at least that's been my experience authoring these things.

But if you can find a way to harmonise them nicely so we get a proper static HTML site at ipld.io _and_ working GitHub links (and also working editor links in VS Code and friends) then that'd be awesome!

**Comment by BigLep** [comment]
@tabcat : contribution is welcome.  For now, we're giving priority to website viewing rather than markdown viewing given we don't know of an easy way to resolve this :(

**Comment by rvagg** [comment]
Technical solution is provided by @dboreham and merged @ #232, the remaining item here is to work through the existing markdown files and rewrite their links so they all work. #232 only did it for a small subset.

**Comment by tabcat** [comment]
Would like to write a script for this sometime in the near future. Seems like there are two cases to make edits:

- add `index.md` to links to folders
- add `.md` to links to files
- may need to adjust path relativity?
